modules/security/src/test/java/org/apache/ignite/internal/processor/security/AbstractSecurityTest.java

```diff
@@ -0,0 +1,231 @@
+        return startGrid(login, "", prmSet, false);
+    }
+
+    /**
+     * @param login Login.
+     * @param pwd Password.
+     * @param prmSet Security permission set.
+     */
+    protected IgniteEx startGrid(String login, String pwd, SecurityPermissionSet prmSet) throws Exception {
+        return startGrid(login, pwd, prmSet, false);
+    }
+
+    /**
+     * @param login Login.
+     * @param prmSet Security permission set.
+     * @param isClient Is client.
+     */
+    protected IgniteEx startGrid(String login, SecurityPermissionSet prmSet,
+        boolean isClient) throws Exception {
+        return startGrid(
+            getConfiguration(login, login, "", prmSet).setClientMode(isClient)
+        );
+    }
+
+    /**
+     * @param login Login.
+     * @param pwd Password.
+     * @param prmSet Security permission set.
+     * @param isClient Is client.
+     */
+    protected IgniteEx startGrid(String login, String pwd, SecurityPermissionSet prmSet,
+        boolean isClient) throws Exception {
+        return startGrid(
+            getConfiguration(login, login, pwd, prmSet).setClientMode(isClient)
+        );
+    }
+
+    /**
+     * @param instanceName Instance name.
+     * @param login Login.
+     * @param pwd Password.
+     * @param prmSet Security permission set.
+     */
+    protected IgniteEx startGrid(String instanceName, String login, String pwd,
+        SecurityPermissionSet prmSet) throws Exception {
+        return startGrid(getConfiguration(instanceName, login, pwd, prmSet));
+    }
+
+    /**
+     * @param instanceName Instance name.
+     * @param login Login.
+     * @param pwd Password.
+     * @param prmSet Security permission set.
+     * @param isClient If true then client mode.
+     */
+    protected IgniteEx startGrid(String instanceName, String login, String pwd,
+        SecurityPermissionSet prmSet, boolean isClient) throws Exception {
+        return startGrid(getConfiguration(instanceName, login, pwd, prmSet).setClientMode(isClient));
+    }
+
+    /**
+     * Getting security permission set builder.
+     */
+    protected SecurityPermissionSetBuilder builder() {
+        return SecurityPermissionSetBuilder.create().defaultAllowAll(true);
+    }
+
+    /**
+     * Getting allow all security permission set.
+     */
+    protected SecurityPermissionSet allowAllPermissionSet() {
+        return builder().build();
+    }
+
+    /**
+     * Method {@link TestRunnable#run()} should throw {@link SecurityException}.
+     *
+     * @param r Runnable.
+     */
+    protected void forbiddenRun(TestRunnable r) {
+        forbiddenRun(r, SecurityException.class);
+    }
+
+    /**
+     * @param r Runnable.
+     * @param types Array of expected exception types.
+     */
+    protected void forbiddenRun(TestRunnable r, Class... types) {
+        try {
+            r.run();
+
+            fail("Should not happen.");
```

Comment:
Let's change it to `Test should throw one of the following exceptions [types.toString]` - to make failure detection easier.